tests/python/gaia-ui-tests/gaiatest/gaia_graphics_test.py

```diff
@@ -48,14 +48,18 @@ def tearDown(self):
         """At the end of test execution, it checks for the errors"""
         self.assertTrue(self.test_passed, msg=self.failcomment)
 
-    def take_screenshot(self):
+    def take_screenshot(self, browser=False, browserInstance=None):
```

Comment:
Also what I noticed is that after invoking take_screenshot() method, it also loses the context to the browser object as well as the player object.  I'm guessing the test case class you mention refers to the actual test case script, (test_browser_play_video.py) then above method might give me an error of 'Element haven't seen before' for the browser object.  that is why I used the browser object to switch to system frame within the take_screenshot method, such as : 
        if browserinstance is None:
            # if the status bar is visible, crop it off
            current_frame = self.marionette.get_active_frame()
            self.marionette.switch_to_frame()
        else:
            browserinstance.switch_to_chrome()
If there is a cleaner way of doing this please let me know.  I did simplify the take_screenshot method per Martijn's suggestion.